gp/cartesian_graph.py

```diff
@@ -4,60 +4,66 @@
 
 try:
     import sympy
+    from sympy.core import expr as sympy_expr
```

Comment:
why not `import sympy.core` and then use `sympy.core.expr`? it's not much longer and it's much clearer that you accessing a member of the sympy package. e.g., below i was wondering whether `sympy_expr` is something we introduced ;)

Reply:
yes, thanks, done.